consensus/src/dag/payload/manager.rs

```diff
@@ -0,0 +1,160 @@
+use super::{
+    payload_fetcher::PayloadRequester,
+    store::{DagPayloadStore, DagPayloadStoreError},
+};
+use crate::dag::{dag_store::DagStore, types::DagPayload, CertifiedNode};
+use anyhow::bail;
+use aptos_collections::BoundedVecDeque;
+use aptos_consensus_types::{
+    common::Payload,
+    dag_payload::{DecoupledPayload, PayloadDigest},
+};
+use aptos_logger::{debug, error};
+use aptos_types::transaction::SignedTransaction;
+use dashmap::DashMap;
+use futures::{future::BoxFuture, FutureExt};
+use std::{ops::DerefMut, sync::Arc};
+use tokio::sync::oneshot;
+
+pub trait TDagPayloadResolver: Send + Sync {
+    fn get_payload_if_exists(&self, node: &CertifiedNode) -> Option<Arc<DecoupledPayload>>;
+    fn add_payload(&self, payload: DecoupledPayload) -> anyhow::Result<()>;
+}
+
+pub struct DagPayloadManager {
+    dag_store: Arc<DagStore>,
+    payload_store: Arc<DagPayloadStore>,
+    requester: PayloadRequester,
+    waiters: DashMap<PayloadDigest, BoundedVecDeque<oneshot::Sender<Vec<SignedTransaction>>>>,
```

Comment:
When do you send the payloads to all senders in the queue?

Reply:
why it's a bounded queue?